spec/config-spec.coffee

```diff
@@ -1106,6 +1126,62 @@ describe "Config", ->
         expect(atom.config.get('foo.bar.str', scope: ['.source.js'])).toBe 'omg'
         expect(atom.config.get('foo.bar.str', scope: ['.source.coffee'])).toBe 'ok'
 
+      describe 'when a schema is added after config values have been set', ->
+        schema = null
+        beforeEach ->
+          schema =
+            type: 'object'
+            properties:
+              int:
+                type: 'integer'
+                default: 2
+              str:
+                type: 'string'
+                default: 'def'
+
+        it "the values set respect the new schema", ->
+          expect(atom.config.set('foo.bar.str', 'global')).toBe true
```

Comment:
What do you think about testing the case of a value that is set before the schema is applied but doesn't have an associated property in the applied schema?